<a href="https://colab.research.google.com/github/ArijeetDiwan/Class114/blob/main/class114.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

linear regression:
 it is a regression model that estimate the relations between 1 dependent variable and 1 independent variable using a straight line  both variable should be contitative 

 regression take a group os random variable,thought to be predecting  Y ,and tries to find to find a mathematical relationship betweeen  them 
 
 this relationship is typically in the form of a straight line that best approximate all the indiviusal data point 

In [2]:
# uploading the csv 
from google.colab import files 
data_to_load=files.upload()


Saving data.csv to data.csv


In [3]:
#here we have height and weight of some people.lets plot this with x axis-height , weight- y axis
import pandas as pd 
import plotly.express as px
df=pd.read_csv("data.csv")
height=df["Height"].tolist()
weight=df["Weight"].tolist()
fig=px.scatter(x=height,y=weight)
fig.show()

here we see that both height and weight are related to each other . The more height more the weight

every line has a formula 
y=mx+c


x=the value in x axis (Height),
y=the value in y axis (weight)

m=slope,
c=intercept in y axis 


we will be using hit and try mathed to find the value of m and c  


In [6]:
m=1
c=0
y=[] 
for x in height :
  y_value=m*x+c
  y.append(y_value)
  
  #plot the point
fig=px.scatter(x=height,y=weight)
fig.update_layout(shapes=[dict(
      type='line',
      y0=min(y),y1=max(y),x0=min(height),x1=max(height)
  )])

fig.show() 

look like  we have to change the value of m and c in such a way that the line fit well with scatter plot line .

In [7]:
m=0.95
c=-93
y=[] 
for x in height :
  y_value=m*x+c
  y.append(y_value)
  
  #plot the point
fig=px.scatter(x=height,y=weight)
fig.update_layout(shapes=[dict(
      type='line',
      y0=min(y),y1=max(y),x0=min(height),x1=max(height)
  )])

fig.show() 

here the line is matching with the scatter plot line so 
now with the given value of m and c let us perdict the weight of a person whose height is 250 using the formula 

In [8]:
#y=mx+c is the formula 
#m=0.95 and c=-93(we found it by hit and try method )
x=250
y=m*x+c
print("weight of someone with height 250 is ",y)

weight of someone with height 250 is  144.5


lets find out the slpoe(m) and intercept(c) of the given data and plot it 


In [10]:
import numpy as np
height_array=np.array(height)
weight_array=np.array(weight)
#slope and intersecpt using pre built function of  numpy
m,c=np.polyfit(height_array,weight_array,1)
print(m,c)

0.9600183750530678 -95.66248461433146


In [11]:
import numpy as np
height_array=np.array(height)
weight_array=np.array(weight)
#slope and intersecpt using pre built function of  numpy
m,c=np.polyfit(height_array,weight_array,1)

y=[]

for x in height_array :
  y_value=m*x+c
  y.append(y_value)
  
  #plot the point
fig=px.scatter(x=height_array,y=weight_array)
fig.update_layout(shapes=[dict(
      type='line',
      y0=min(y),y1=max(y),x0=min(height_array),x1=max(height_array)
  )])

fig.show() 


this is the best fit line for the data .the computer algorithm has provide us .
let us find the weight of someone having height 250 using the new value of m  and c (found from               computer algorithm  )

In [12]:
x=250
#y=0.9600183750530678 ,
#c=-95.66248461433146
y=m*x+c
print("weight of someone with height 250 is ",y)


weight of someone with height 250 is  144.3421091489355



CONCLUSION

we have just built to predict the weight of the person base on height 
 earlier we got value 144.5 now 144.34(nearly equal)